syntax = "proto3";

service TimeSeriesService {
  rpc insert (TimeSeries) returns (ReindexingResponse) {}
  rpc query (TimeSeries) returns (TimeSeriesIdx) {}
  rpc reindex (ReindexingRequest) returns (ReindexingResponse) {}
}

message ReindexingResponse {
  int32 status = 1;	
}

message ReindexingRequest {
  int32 n_samples = 1;
}

message TimeSeries {
  repeated float ts = 1;
  int32 length = 2;
  int32 dim = 3;
}

message TimeSeriesIdx {
  repeated int32 ids = 1;
  string messages = 2;
}